packages/nimble-components/src/table/specs/table-columns-hld.md

```diff
@@ -250,38 +313,45 @@ public class TableColumnPositiveNegativeNumber extends TableColumn<TableColumnPo
         return [valueKey];
     }
 
-    public readonly cellStyles: ElementStyles =
-        css`
-            .good {
-                color: green;
-            }
-
-            .bad {
-                color: red;
-            }
-        `;
-
-    public readonly cellTemplate: ViewTemplate<TableCellState<TableColumnPositiveNegativeNumberCellData>> =
-        html<TableCellState<TableColumnPositiveNegativeNumberCellData>`
-            <nimble-text-field
-                class="${x => isPositive(x.data.value) ? "good" : "bad"}"
-                readonly="true"
-                value="${x => x.data.value}"
-            >
-            </nimble-text-field>
-        `;
-
     public validateCellData(cellData: TCellRecord): void {
         if (typeof(cellData['value']) !== 'number') {
             throw new Error('Type for cellData is incorrect!');
         }
     }
 }
+
+class PositiveNegativeNumberCellElement extends BaseCellElement<TableColumnPositiveNegativeNumberCellData> {
+    @volatile
+    public get textFieldCssClass(): string {
+        return this.cellRecord.value > 0 ? 'good' : 'bad';
```

Comment:
For this to work the fields of the cellRecord objects themselves need to be observable I think. Although if we always update the full cellRecord object reference (ie activate the setter for the cellRecord property) then it should be fine. We just can't update the fields directly, we have to create a new object that is assigned to the setter of cellRecord (so it passes a diff check).

Reply:
Pretty sure we will never update the fields of an existing cellRecord.